components/usage/pkg/apiv1/usage.go

```diff
@@ -118,6 +118,105 @@ func (s *UsageService) ListBilledUsage(ctx context.Context, in *v1.ListBilledUsa
 	}, nil
 }
 
+func (s *UsageService) ListUsage(ctx context.Context, in *v1.ListUsageRequest) (*v1.ListUsageResponse, error) {
+	to := time.Now()
+	if in.To != nil {
+		to = in.To.AsTime()
+	}
+	from := to.Add(-maxQuerySize)
+	if in.From != nil {
+		from = in.From.AsTime()
+	}
+
+	if from.After(to) {
+		return nil, status.Errorf(codes.InvalidArgument, "Specified From timestamp is after To. Please ensure From is always before To")
+	}
+
+	if to.Sub(from) > maxQuerySize {
+		return nil, status.Errorf(codes.InvalidArgument, "Maximum range exceeded. Range specified can be at most %s", maxQuerySize.String())
+	}
+
+	var order db.Order = db.DescendingOrder
```

Comment:
I think the type here is rendundant and could be removed 
```suggestion
	order := db.DescendingOrder
```